src/components/API.js

```diff
@@ -0,0 +1,118 @@
+import { initialCards } from "../utils/constants";
+
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    // if the server returns an error, reject the promise
+    return Promise.reject(`Error: ${res.status}`);
+    //throw new Error(`Error: ${res.status}`);
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    }).then((res) => {
+      return this._checkResponse(res);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.

Reply:
Not all of my fetch are the same, I have :
`${this._baseUrl}/cards` 
`${this._baseUrl}/users/me`
`${this._baseUrl}/users/me/avatar`
`${this._baseUrl}/cards/${cardId}/likes`
`${this._baseUrl}/cards/${cardId}`

How would that still work?